### DISCUSSION TITLE
直接focusFirstDescendant是否有点太主观了？

### DISCUSSION THREAD

**Discussion by thankwsx** [tusen-ai/naive-ui#2531] [General]
https://github.com/TuSimple/naive-ui/blob/a868b5033ddd2d3021efeaac13dc035be9d99204/src/modal/src/BodyWrapper.tsx#L213

在dialog.warning里使用RadioGroup组件时，默认focus第一个元素，导致ui上引起误解。

<img width="155" alt="image" src="https://user-images.githubusercontent.com/90209/156321959-513c44e5-597a-4485-97d0-105451c2fcf1.png">

dialog底层调用的modal，没有地方传参数autoFocus为false啊

是不是modal里autoFocus默认值改为false比较好？

现在我是通过在dialog里加一个w-0的input来hack的。

**Comment by 07akioni** [comment]
这个是属于一个价值取向的问题，没有绝对的正确答案。

我倾向是自动聚焦。

不光我是这么弄，chakra ui 也是这么搞的。

![image](https://user-images.githubusercontent.com/18677354/156391374-79220f6e-6a06-4d3f-bf6e-0505ca65fcd8.png)

**Reply by 07akioni** [reply]
我能理解你的意见，但是这个默认选项只能有一个。我倾向于默认 Focus，W3C 的 a11y guideline 也是这样做的。

**Reply by thankwsx** [reply]
的确是一个主观的细节表现。
但是针对我截图的具体情况，我理解也要考虑具体场景。
input text这种文本输入框，aotufocus有点实际意义。

个人觉得，要么只对text进行autofocus
或者ui组件库（对，就是naive-ui），对focus和checked的状态设计一个比较明显的识别度。（估计很难）

**Reply by 07akioni** [reply]
你讲的道理我是理解的，但是这是个选择的问题，只是恰好这个库的维护者认为这样会更好。

确实是个甜豆浆和咸豆浆的问题。
